server/apps/immich/src/api-v1/asset/dto/create-asset.dto.ts

```diff
@@ -32,6 +33,12 @@ export class CreateAssetDto {
 
   @IsOptional()
   duration?: string;
+
+  @ApiProperty({ type: 'string', format: 'binary' })
```

Comment:
Can we just add validation right here itself, or is there some reason this doesn't work?

Reply:
When I add `@IsNotEmpty` to this it will always say the field is empty, also when it isn't. Probably because the FileFieldsInterceptor has already taken the file. The reason I even added this field, is for the API generation to work properly. When I don't add this `assetData` doesn't show up in any generated code or documentation.

Reply:
I was wondering if something like that would happen, makes sense. Might be worth adding a comment to indicate it's open for open-api only and that validation is run else where.